src/app/map/[id]/page.tsx

```diff
@@ -1,103 +1,112 @@
 import { adapt, HexTileData } from "./State/types";
-import {
-  HexCoord,
-  HexDirection,
-} from "~/lib/domains/mapping/utils/hex-coordinates";
-import { DEFAULT_HEXMAP_COLORS } from "~/lib/domains/mapping/_objects/hex-map";
+import { CoordSystem } from "~/lib/domains/mapping/utils/hex-coordinates";
 
 interface HexMapPageProps {
-  params: { id: string };
-  searchParams: {
+  params: Promise<{ id: string }>;
+  searchParams: Promise<{
     scale?: string;
     expandedItems?: string;
     isDynamic?: string;
     focus?: string;
-  };
+  }>;
 }
 
 export default async function HexMapPage({
   params,
   searchParams,
 }: HexMapPageProps) {
-  const { id: mapId } = await params;
+  const { id: rootItemId } = await params;
   const searchParamsString = await searchParams;
 
-  const { data: center } = await getMapData(mapId);
+  const { data: rootItem } = await getRootItemData(rootItemId);
 
-  if (!center || !center.center) {
+  if (!rootItem) {
     return notFound();
   }
 
-  // Handle initial focus parameter
+  // Handle initial focus parameter - default to root item coordinates
   if (!searchParamsString.focus) {
-    const defaultFocus = center.center.coordinates;
-    const currentPath = `/map/${mapId}`;
-    const newSearchParams = new URLSearchParams(searchParamsString as any); // Cast to any to allow adding new params
+    const defaultFocus = rootItem.coordinates;
+    const currentPath = `/map/${rootItemId}`;
+    const newSearchParams = new URLSearchParams(searchParamsString as any);
     newSearchParams.set("focus", defaultFocus);
     return redirect(`${currentPath}?${newSearchParams.toString()}`);
   }
 
-  const { data: items, error: itemsError } = await getMapItems(mapId);
-
-  const { pathname, currentSearchParamsString: currentSearchParamsWithFocus } =
-    safeGetPathnameAndSearchParams(searchParamsString, mapId);
+  const { data: items, error: itemsError } = await getMapItems(rootItemId);
 
   if (itemsError || !items) {
-    return <null; //<Loading />;>
+    return null; // TODO: Re-enable Loading component when ready
   }
 
-  const { scale, expandedItemIds, isDynamic, focus } = initMapParameters(
+  const { scale, expandedItemIds } = initMapParameters(
     searchParamsString,
     items,
   );
 
   const mapItems = formatItems(items);
+
+  // Extract userId and groupId from root item coordinates for components that need them
+  const rootCoords = CoordSystem.parseId(rootItem.coordinates);
+
   return (
     <div className="relative flex h-full w-full flex-col">
       <StaticMapCanvas
-        center={center.center.coordinates}
+        center={rootItem.coordinates}
         items={mapItems}
         scale={scale}
         expandedItemIds={expandedItemIds}
-        pathname={pathname}
-        currentSearchParamsString={currentSearchParamsWithFocus}
+        rootItemId={parseInt(rootItemId)}
+        userId={rootCoords.userId}
+        groupId={rootCoords.groupId}
```

Comment:
When I previously talked about drilling down a prop URLInfo I think it will start here